--- stock/garglk/cggestal.c	2011-10-10 15:16:12.000000000 -0700
+++ garglk-edited/garglk/cggestal.c	2012-09-03 04:08:29.424253856 -0700
@@ -102,7 +102,10 @@
             if (val == wintype_Graphics)
                 return gli_conf_graphics;
             return FALSE;
-
+	case gestalt_DrawSVG:
+	    if (val == wintype_TextBuffer)
+		return gli_conf_svg;
+	    return FALSE;
         case gestalt_Sound:
         case gestalt_SoundVolume:
         case gestalt_SoundMusic:
