--- stock/garglk/garglk.h	2011-10-10 15:14:40.000000000 -0700
+++ garglk-edited/garglk/garglk.h	2012-09-04 13:43:58.465812860 -0700
@@ -219,6 +219,7 @@
 extern int gli_conf_lcd;
 
 extern int gli_conf_graphics;
+extern int gli_conf_svg;
 extern int gli_conf_sound;
 extern int gli_conf_speak;
 
@@ -712,6 +714,7 @@
 void win_graphics_set_background_color(window_graphics_t *cutwin, glui32 color);
 
 glui32 win_textbuffer_draw_picture(window_textbuffer_t *dwin, glui32 image, glui32 align, glui32 scaled, glui32 width, glui32 height);
+glui32 win_textbuffer_draw_unscaled_pic(window_textbuffer_t *dwin, picture_t *pic, glui32 align);
 glui32 win_textbuffer_flow_break(window_textbuffer_t *win);
 
 void gli_calc_padding(window_t *win, int *x, int *y);
