diff -Naur old/gi_dispa.c new/gi_dispa.c
--- gi_dispa.c	2012-10-31 12:02:19.000000000 -0700
+++ gi_dispa.c	2012-08-25 12:27:06.000000000 -0700
@@ -247,6 +247,7 @@
     { 0x00E0, glk_image_get_info, "image_get_info" },
     { 0x00E1, glk_image_draw, "image_draw" },
     { 0x00E2, glk_image_draw_scaled, "image_draw_scaled" },
+    { 0x00E3, glk_svg_draw, "svg_draw" },
     { 0x00E8, glk_window_flow_break, "window_flow_break" },
     { 0x00E9, glk_window_erase_rect, "window_erase_rect" },
     { 0x00EA, glk_window_fill_rect, "window_fill_rect" },
@@ -524,6 +525,8 @@
             return "5QaIuIsIs:Iu";
         case 0x00E2: /* image_draw_scaled */
             return "7QaIuIsIsIuIu:Iu";
+	case 0x00E3: /* svg_draw */
+	    return "3QaUIs:";
         case 0x00E8: /* window_flow_break */
             return "1Qa:";
         case 0x00E9: /* window_erase_rect */
@@ -1053,6 +1056,11 @@
                 arglist[2].sint, arglist[3].sint,
                 arglist[4].uint, arglist[5].uint);
             break;
+        case 0x00E3: /* svg_draw */
+            glk_svg_draw(arglist[0].opaqueref,
+                arglist[1].unicharstr,
+                arglist[2].sint);
+            break;
         case 0x00E8: /* window_flow_break */
             glk_window_flow_break(arglist[0].opaqueref);
             break;
diff -Naur old/glk.h new/glk.h
--- glk.h	2012-10-31 12:02:19.000000000 -0700
+++ glk.h	2012-08-25 12:28:58.000000000 -0700
@@ -364,6 +364,7 @@
 #define imagealign_MarginRight (0x05)
 
 extern glui32 glk_image_draw(winid_t win, glui32 image, glsi32 val1, glsi32 val2);
+extern glui32 glk_svg_draw(winid_t win, glui32 *svg_string, glui32 align);
 extern glui32 glk_image_draw_scaled(winid_t win, glui32 image,
     glsi32 val1, glsi32 val2, glui32 width, glui32 height);
 extern glui32 glk_image_get_info(glui32 image, glui32 *width, glui32 *height);
