--- stock/garglk/window.c	2011-10-06 15:59:54.000000000 -0700
+++ garglk-edited/garglk/window.c	2012-09-05 10:15:07.868321622 -0700
@@ -21,14 +21,19 @@
  *                                                                            *
  *****************************************************************************/
 
+#include <assert.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include "glk.h"
 #include "garglk.h"
+#include <librsvg/rsvg.h>
+#include <librsvg/rsvg-cairo.h>
+#include <cairo.h>
 
 #define LINES 24
 #define COLS 70
+#define DEBUG 0
 
 int gli_force_redraw = 1;
 int gli_more_focus = 0;
@@ -1322,8 +1327,124 @@
 }
 
 /*
- * Graphics and Image drawing
- */
+* Graphics and Image drawing
+*/
+
+static void fail(char *msg) {
+    printf ("SVG Error: %s\n", msg);
+    if(!DEBUG)
+        exit (-1);
+}
+
+picture_t *glk_svg_to_pic(winid_t win, char *svg_string) {
+    picture_t *pic = malloc(sizeof(*pic));
+    int rgbaSize;
+    unsigned char* color;
+    GError *error = NULL;
+    RsvgHandle *handle;
+    RsvgDimensionData dim;
+    cairo_surface_t *surface;
+    cairo_t *cr;
+    cairo_status_t status;
+    guchar *buf = (guchar*) svg_string;
+
+    assert(pic);
+    g_type_init ();
+    rsvg_set_default_dpi (72.0);
+    handle = rsvg_handle_new_from_data (buf, strlen(svg_string), &error);
+    if(error != NULL) {
+ 	fail (strcat(error->message, svg_string));     
+	return NULL;
+    }
+
+    rsvg_handle_get_dimensions (handle, &dim);
+    surface = cairo_image_surface_create (CAIRO_FORMAT_ARGB32, dim.width, dim.height);
+    cr = cairo_create (surface);
+
+    rsvg_handle_render_cairo (handle, cr);
+    status = cairo_status (cr);
+    if (status) {
+	fail (strcat((char * const) cairo_status_to_string (status),
+                     svg_string));
+	return NULL;
+    }
+
+    cairo_surface_flush(surface);
+    pic->refcount = 1;
+    pic->w = dim.width;
+    pic->h = dim.height;
+    rgbaSize = pic->w * pic->h * 4;
+    pic->rgba = malloc(rgbaSize);
+    assert(pic->rgba);
+    memcpy(pic->rgba, cairo_image_surface_get_data(surface), rgbaSize);
+    pic->id = 0;
+    pic->scaled = 0;
+    cairo_destroy (cr);
+    cairo_surface_destroy (surface);
+    return pic;
+}
+
+glui32 glk_svg_draw(winid_t win, glui32 *svg_string, glui32 align, glui32 val2)
+{
+    if (!gli_conf_svg)
+        return FALSE;
+
+    if(DEBUG)
+        glk_put_string_uni(svg_string);
+
+    int svgBodyLen = strlen_uni(svg_string);
+    char svgBody[svgBodyLen + 1];
+    char *svgBodyP = svgBody;
+    char svgHeader [] = "<?xml version='1.0' standalone='no'?><!DOCTYPE svg PUBLIC '-//W3C//DTD SVG 1.2//EN' 'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'><svg xmlns='http://www.w3.org/2000/svg' version='1.1'>";
+    char svgFooter [] = "</svg>";
+    char svg[strlen(svgHeader) + svgBodyLen + strlen(svgFooter) + 1];
+    picture_t *renderedPic;
+
+    while(svgBodyLen--) 
+        *svgBodyP++ = *svg_string++;
+    *svgBodyP++ = '\0';
+
+    strcpy(svg, svgHeader);
+    strcat(svg, svgBody);
+    strcat(svg, svgFooter);
+
+    renderedPic = glk_svg_to_pic(win, svg);
+    if(renderedPic == NULL) 
+	return FALSE;
+    win_textbuffer_draw_unscaled_pic(win->data, renderedPic, align);
+    return TRUE;
+}
+
+glui32 glk_svg_draw_scaled(winid_t win, glui32 *svg_string, glui32 align, glui32 val2, glui32 width, glui32 height)
+{
+    if (!gli_conf_svg)
+        return FALSE;
+ 
+    if(DEBUG)
+        glk_put_string_uni(svg_string);
+
+    int svgBodyLen = strlen_uni(svg_string);
+    char svgBody[svgBodyLen + 1];
+    char *svgBodyP = svgBody;
+    char svgHeader [] = "<?xml version='1.0' standalone='no'?><!DOCTYPE svg PUBLIC '-//W3C//DTD SVG 1.2//EN' 'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'><svg xmlns='http://www.w3.org/2000/svg' version='1.1'>";
+    char svgFooter [] = "</svg>";
+    char svg[strlen(svgHeader) + svgBodyLen + strlen(svgFooter) + 1];
+    picture_t *renderedPic;
+
+    while(svgBodyLen--)
+        *svgBodyP++ = *svg_string++;
+    *svgBodyP++ = '\0';
+
+    strcpy(svg, svgHeader);
+    strcat(svg, svgBody);
+    strcat(svg, svgFooter);
+
+    renderedPic = glk_svg_to_pic(win, svg);
+    if(renderedPic == NULL)
+        return FALSE;
+    win_textbuffer_draw_picture(win->data, renderedPic, align, TRUE, width, height);
+    return TRUE;
+}
 
 glui32 glk_image_draw(winid_t win, glui32 image, glsi32 val1, glsi32 val2)
 {
